tests/unit/test_tracing.c

```diff
@@ -1097,6 +1097,19 @@ SENTRY_TEST(txn_data)
     sentry__transaction_decref(txn);
 }
 
+SENTRY_TEST(ctx_trace_data)
+{
+    sentry_transaction_t *txn
+        = sentry__transaction_new(sentry_value_new_object());
+
+    sentry_transaction_set_data(
+        txn, "os.name", sentry_value_new_string("Linux"));
+    check_after_set(txn->inner, "extra", "os.name", "Linux");
+    sentry_value_set_by_key(txn->inner, "sampled", sentry_value_new_bool(true));
+
+    sentry_transaction_finish(txn);
```

Comment:
We need to add an assertion here (or else there is no test); how can we assert that the context.trace.data actually holds the data from our transaction?

Reply:
You can't do it with the current unit tests. At one point, we could introduce a "test transport" where we assert all the things happening in the pre-send pipeline.

In the meantime, we capture these with our integration tests (like the one currently failing). Writing tests for unpacking nested structures is much more convenient to express (and thus readable) in Python than in C (achievable with a little string and macro magic). 

Test execution speed is also a factor that hints towards having unit-test access to the transport pipeline.